### ISSUE TITLE
EOF when doing pachctl list file 

### ISSUE THREAD

**Issue by pappasilenus** [pachyderm/pachyderm#4778]
**Raanan Hadar** 4/8/2020 3:29 pm
I encountered a weird bug in azure that I wanted to share, luckily the reproduction is relatively simple.  
  
\---  
  
Ticket created from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586273553185700) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) in #users.

---

**John Karabaic** 4/8/2020 3:30 pm *(private)*
I have an input repo with videos and a simple '/\*' pipeline where i take each video and save its frames as output using opencv.

* * *

Note added from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586273595186600) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) ([jk@pachyderm.io](mailto:jk@pachyderm.io)) in [#users](https://pachyderm-users.slack.com/messages/C3UN0S0Q0).

---

**John Karabaic** 4/8/2020 3:30 pm *(private)*
\`\`\`import cv2  
import os

this version is fine
====================

def video\_to\_frames(video):  
cap = cv2.VideoCapture(video)  
video\_name, ext = os.path.splitext(os.path.split(video)\[1\])  
success = True  
i = 0  
while success:  
success, frame = cap.read()  
dest\_path = os.path.join("/pfs/out", video\_name + '\_' + str(i)+'.jpg')  
cv2.imwrite(dest\_path,frame)  
print("wrote frame:", str(i), "to:", dest\_path)  
i += 1  
cap.release()

this version produces EOFs on Azure
===================================

def video\_to\_frames(video):  
cap = cv2.VideoCapture(video)  
video\_name, ext = os.path.splitext(os.path.split(video)\[1\])  
os.mkdir(os.path.join("/pfs/out", video\_name))  
success = True  
i = 0  
while success:  
success, frame = cap.read()  
dest\_path = os.path.join("/pfs/out", video\_name, str(i)+'.jpg')  
cv2.imwrite(dest\_path,frame)  
print("wrote frame:", str(i), "to:", dest\_path)  
i += 1  
cap.release()

for dirpath, dirs, files in os.walk("/pfs/in\_repo"):  
for file in files:  
video\_to\_frames(os.path.join(dirpath, file))\`\`\`

* * *

Note added from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586273817190200?thread_ts=1586273817.190200&cid=C3UN0S0Q0) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) ([jk@pachyderm.io](mailto:jk@pachyderm.io)) in [#users](https://pachyderm-users.slack.com/messages/C3UN0S0Q0).

---

**John Karabaic** 4/8/2020 3:30 pm *(private)*
just to clarify, i'm not seeing the EOF in the logs, i'm seeing it when doing a pachctl list file

* * *

Note added from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586274809194000?thread_ts=1586273817.190200&cid=C3UN0S0Q0) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) ([jk@pachyderm.io](mailto:jk@pachyderm.io)) in [#users](https://pachyderm-users.slack.com/messages/C3UN0S0Q0).

---

**John Karabaic** 4/8/2020 3:31 pm *(private)*
i see many "ERROR PPS master: error retrieving RC for "vid2frame": RC not found; retrying in 648.595633ms"

* * *

Note added from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586275376196300?thread_ts=1586273817.190200&cid=C3UN0S0Q0) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) ([jk@pachyderm.io](mailto:jk@pachyderm.io)) in [#users](https://pachyderm-users.slack.com/messages/C3UN0S0Q0).

---

**John Karabaic** 4/8/2020 3:31 pm *(private)*
and when i list the files i see

* * *

Note added from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586275386196500?thread_ts=1586273817.190200&cid=C3UN0S0Q0) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) ([jk@pachyderm.io](mailto:jk@pachyderm.io)) in [#users](https://pachyderm-users.slack.com/messages/C3UN0S0Q0).

---

**John Karabaic** 4/8/2020 3:31 pm *(private)*
ERROR pfs.API.ListFileStream {"duration":0.062201601,"error":"unexpected EOF","request":{"file":{"commit":{"repo":{"name":"vid2frame"},"id":"d5ee1ea618994272ac6c35efce997622"},"path":"/data"}},"response":"response stream with 436 objects"} \[\]

* * *

Note added from a Slack [message](https://pachyderm-users.slack.com/archives/C3UN0S0Q0/p1586275425196700?thread_ts=1586273817.190200&cid=C3UN0S0Q0) by [John Karabaic (he/him)](https://pachyderm-users.slack.com/team/UD5JT8JRE) ([jk@pachyderm.io](mailto:jk@pachyderm.io)) in [#users](https://pachyderm-users.slack.com/messages/C3UN0S0Q0).
 gz#16

**Comment by RaananHadar** [comment]
Just to update on something interesting. I tried creating a pipeline that performs a groupby operation on the results of the above pipeline that is 'fine'. The pipeline performs an empty_files: true shuffle operation by creating a symlink to directories having the original movie names. The Job itself fails with unexpected EOF. I believe this is related but may allow to isolate the problem better. 

If needed, I can provide you with the data a code to reproduce everything.

**Comment by RaananHadar** [comment]
So just to update on my personal journey with this. 

I reinstalled Pachyderm on AKS using [minio azure S3 gateway](https://docs.min.io/docs/minio-gateway-for-azure.html) feature to mask for Azure lack of compliance with S3. Using the same storage account as before, the former pipeline seems to be working now so far. I will keep you updated if this returns, but **the current install instruction on azure are broken for me at least**.

I am documenting here some yaml files which I wrote for deploying minio with azure gateway functionality inside aks, because these files are not in the official minio docs don't have any yaml files for this scenario, I assume that many azure users can benefit from them:

Here is the minio deployment:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  # This name uniquely identifies the Deployment
  name: minio-gw
spec:
  selector:
    matchLabels:
      app: minio # has to match .spec.template.metadata.labels
  strategy:
    # Specifies the strategy used to replace old Pods by new ones
    # Refer: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
    type: Recreate
  template:
    metadata:
      labels:
        # This label is used as a selector in Service definition
        app: minio
    spec:
      containers:
      - name: minio
        # Pulls the lastest Minio image from Docker Hub
        image: minio/minio:RELEASE.2020-04-15T19-42-18Z
        args:
        - gateway
        - azure
        env:
        # MinIO access key and secret key
        - name: MINIO_ACCESS_KEY
          value: <your_access_key_goes_here>
        - name: MINIO_SECRET_KEY
          value: <your_secret_key_goes_here>
        ports:
        - containerPort: 9000
        # Readiness probe detects situations when MinIO server instance
        # is not ready to accept traffic. Kubernetes doesn't forward
        # traffic to the pod while readiness checks fail.
        readinessProbe:
          httpGet:
            path: /minio/health/ready
            port: 9000
          initialDelaySeconds: 120
          periodSeconds: 20
        # Liveness probe detects situations where MinIO server instance
        # is not working properly and needs restart. Kubernetes automatically
        # restarts the pods if liveness checks fail.
        livenessProbe:
          httpGet:
            path: /minio/health/live
            port: 9000
          initialDelaySeconds: 120
          periodSeconds: 20
```
and the service:
```
apiVersion: v1
kind: Service
metadata:
  name: minio-service
spec:
  ports:
    - port: 9000
      targetPort: 9000
      protocol: TCP
  selector:
    app: minio
```
And I deployed pachyderm using the custom deploy:
1. use the following pachctl deploy command:
```
pachctl deploy custom --persistent-disk aws --object-store s3 foobar 10 <pachyderm_bucket_name>  'your_access_key_goes_here' 'your_secret_key_goes_here' 'http://minio-service:9000' --dynamic-etcd-nodes 1 --local-roles --output yaml --dry-run > custom_deploy.yaml
```
2. I changed the storage class in the custom_deploy.yaml file to 'managed-premium' which will properly allocated a dynamic persistent volume in your AKS cluster.

**Comment by pappasilenus** [comment]
@nitinjainsj I have a cluster set up where I can reproduce this issue. @ me for info.

**Comment by pappasilenus** [comment]
Agent comment from John Karabaic in <a href='https://pachyderm.zendesk.com/agent/tickets/16'>Zendesk ticket #16</a>:<br><br>Hi, Raanan,

This was fixed in 1.10.3.  Can you confirm the fix on your side?

**Comment by RaananHadar** [comment]
Will do.

**Comment by pappasilenus** [comment]
Agent comment from Raanan Hadar in <a href='https://pachyderm.zendesk.com/agent/tickets/16'>Zendesk ticket #16</a>:<br><br>hi, will set up an extra cluster and test this on my env. will update soon.

**Comment by pappasilenus** [comment]
Agent comment from Raanan Hadar in <a href='https://pachyderm.zendesk.com/agent/tickets/16'>Zendesk ticket #16</a>:<br><br>As I noted in the github issue, I tested this and it was indeed solved in 1.10.3 !
